nevergrad/optimization/test_optimizerlib.py

```diff
@@ -137,6 +137,44 @@ def check_optimizer(
 UNSEEDABLE: tp.List[str] = []
 
 
+def buggy_function(x: np.ndarray) -> float:
+    if any(x[::2] > 0.0):
+        return float("nan")
+    if any(x > 0.0):
+        return float("inf")
+    return np.sum(x ** 2)
+
+
+@skip_win_perf  # type: ignore
+@pytest.mark.parametrize("name", registry)  # type: ignore
+def test_infnan(name: str) -> None:
+    optim_cls = registry[name]
+    if (
+        "Cobyla" not in name  # Scipy doesn't like nan or plateaus..
+        and "Powell" not in name
+        and "SQP" not in name
+        and "EDA" not in name  # Those ones rely on specific recommendations... could be fixed though
+        and "EMNA" not in name  # Those ones rely on specific recommendations... could be fixed though
+        and "CMandAS3" != name
+        and "CMA" not in name
+        and "Stupid" not in name  # Does not have to do anything meaningful
+        and "Large" not in name  # Dedicated to problems with optimum far away
+        and "TBPSA" not in name  # Dedicated to noisy optimization
+        and "SPSA" != name  # Dedicated to noisy optimization
+        and "BO" not in name  # Our BO does weird things sometimes
+        and "Noisy" not in name  # Dedicated to noisy optimization
+        and "chain" not in name  # Sometimes chaining is not adapted, let us remove all of them
+        and "NGOptBase" != name
+        and "Shiwa" != name
+        and "NGOpt2" != name
+        and "NGO" != name
+    ):
```

Comment:
please factorize, I'm pretty sure this is going to be worse and worse so better do it early
you could just have one list of names with and then do `if any(x in exclude)`
or two lists if you want to have exact matches for some.

Then again, that would still be hard to maintain, so whenever possible it should actually test with `isinstance(optim, excluded_classes)`
Why is it better?
- because you get all the child classes with it, whatever their name
- because if names change, then you'll have to modify the code here (with strings... we may change name, and leave some strings here for eternity and beyond for no reasons)
I understand it's not always possible though (for some algorithms, you only know it uses BO under the hood because of the name...)

Reply:
Simplified a lot.
Actually we should remove a lot of old optimizers.
This is impossible for now due to the "mixsimulator" import which crashes if the list of optimization methods is reduced --- this should be fixed soon.